Wrappers/Python/ccpi/framework/framework.py

```diff
@@ -113,14 +130,51 @@ def __str__ (self):
         repres += "voxel_size : x{0},y{1},z{2}\n".format(self.voxel_size_x, self.voxel_size_y, self.voxel_size_z)
         repres += "center : x{0},y{1},z{2}\n".format(self.center_x, self.center_y, self.center_z)
         return repres
-    def allocate(self, value=0, dimension_labels=None):
+    
+    def allocate(self, value=None, dimension_labels=None):
```

Comment:
this should be implemented like this:
```python
def allocate(self, value=0, dimension_labels=None):
        '''allocates an ImageData according to the size expressed in the instance'''
        if dimension_labels is None:
            dimension_labels = self.dimension_labels
        out = ImageData(geometry=self, dimension_labels=dimension_labels)
        if isinstance(value, Number):
            if value != 0:
                out += value
        else:
            if value == 'random':
                out.fill(numpy.random.random_sample(out.shape))
            elif value == 'random_int':
                out.fill(numpy.random.random_int(10, out.shape))
        return out
```

Reply:
Can we have a random.seed for random, random_int?